src/rhsmlib/facts/network.py

```diff
@@ -0,0 +1,176 @@
+            "network.ipv4_address": ", ".join(self._extract_address_list("inet", data)),
+            "network.ipv6_address": ", ".join(self._extract_address_list("inet6", data)),
+        }
+        return result
+
+    def get_interfaces(self) -> dict:
+        """Get detailed network interface facts.
+
+        Interface names, IPv4/IPv6 addresses and masks get collected here.
+
+        Resulting facts have 'net.' prefix.
+        """
+        data = self._query_ip_command()
+        result: Dict[str, Union[str, int]] = {}
+
+        for interface in data:
+            prefix: str = f"net.interface.{interface['ifname']}"
+
+            # MAC address
+            if "LOOPBACK" in interface["flags"] or "NOARP" in interface["flags"]:
+                # Loopback has a MAC address of '00:00:00:00:00:00'.
+                # Tunnels have their MAC randomized every time, see BZ#838123.
+                pass
+            else:
+                result[f"{prefix}.mac_address"] = interface["address"]
+                if "permaddr" in interface:
+                    # Wireless interfaces have permanent address and temporary
+                    # address they use to identify to unknown access points.
+                    result[f"{prefix}.permanent_mac_address"] = interface["permaddr"]
+
+            # IP address
+            ipv4_addresses: List[str] = []
+            ipv4_broadcasts: List[str] = []
+            ipv4_netmasks: List[int] = []
+
+            ipv6_global_addresses: List[str] = []
+            ipv6_link_addresses: List[str] = []
+            ipv6_host_addresses: List[str] = []
+            ipv6_global_netmasks: List[int] = []
+            ipv6_link_netmasks: List[int] = []
+            ipv6_host_netmasks: List[int] = []
+
+            for address in interface["addr_info"]:
+                if address["family"] == "inet":
+                    ipv4_addresses.append(address["local"])
+                    # Localhost does not have a broadcast address
+                    if "broadcast" in address:
+                        ipv4_broadcasts.append(address["broadcast"])
+                    else:
+                        # FIXME Should localhost's broadcast be simply omitted?
+                        ipv4_broadcasts.append("Unknown")
+                    ipv4_netmasks.append(address["prefixlen"])
+
+                if address["family"] == "inet6":
+                    if address["scope"] == "global":
+                        ipv6_global_addresses.append(address["local"])
+                        ipv6_global_netmasks.append(address["prefixlen"])
+                    if address["scope"] == "link":
+                        ipv6_link_addresses.append(address["local"])
+                        ipv6_link_netmasks.append(address["prefixlen"])
+                    if address["scope"] == "host":
+                        ipv6_host_addresses.append(address["local"])
+                        ipv6_host_netmasks.append(address["prefixlen"])
```

Comment:
minor optimization: i'd use an `if ... elif ...` chain for this `if address["family"] == "inet6"` and all the `if`s inside it, since each condition is mutually exclusive with the others (sadly we cannot use the switch statement, available in python 3.10+)